File: NAP II - POO/Python/UFRAcommerce/endereco.py
class Endereco:
    def __init__(self, nome, logradouro, numero, complemento=""):
        self.nome = nome
        self.logradouro = logradouro
        self.numero = numero
        self.complemento = complemento

    def __str__(self):
        return f"Endereco(nome={self.nome}, logradouro={self.logradouro}, numero={self.numero}, complemento={self.complemento})"

File: NAP II - POO/Java/src/Endereco.java
public class Endereco {
    private String logradouro;
    private String numero;
    private String complemento;
    private String nome;

    // Construtor padrão
    public Endereco(String logradouro, String numero, String complemento, String nome) {
        this.logradouro = logradouro;
        this.numero = numero;
        this.complemento = complemento;
        this.nome = nome;
    }

    // Construtor não padrão
    public Endereco(String logradouro, String numero, String nome) {
        this(logradouro, numero, "Sem complemento", nome);
    }

    @Override
    public String toString() {
        return String.format("Endereço: %s, %s, %s, %s", logradouro, numero, complemento, nome);
    }
}
